From: Steve Dickson <SteveD@redhat.com>
Date: Mon, 1 Feb 2010 20:46:03 -0500
Subject: [nfs] Try to commit unstable writes in nfs_release_page()
Message-id: <4B673D8B.5090708@RedHat.com>
Patchwork-id: 23062
O-Subject: [RHEL6] [PATCH 2/8] NFS: Try to commit unstable writes in
	nfs_release_page()
Bugzilla: 560784
RH-Acked-by: Peter Staubach <staubach@redhat.com>
RH-Acked-by: Jeff Layton <jlayton@redhat.com>

commit 82be934a59ff891cac598727e5a862ba2b9d1fac
Author: Trond Myklebust <Trond.Myklebust@netapp.com>
Date:   Tue Jan 26 15:41:53 2010 -0500

    NFS: Try to commit unstable writes in nfs_release_page()

    If someone calls nfs_release_page(), we presumably already know that the
    page is clean, however it may be holding an unstable write.

    Signed-off-by: Trond Myklebust <Trond.Myklebust@netapp.com>
    Cc: stable@kernel.org
    Reviewed-by: Chuck Lever <chuck.lever@oracle.com>

Signed-off-by: Aristeu Rozanski <arozansk@redhat.com>

diff --git a/fs/nfs/file.c b/fs/nfs/file.c
index f5fdd39..393d40f 100644
--- a/fs/nfs/file.c
+++ b/fs/nfs/file.c
@@ -486,6 +486,8 @@ static int nfs_release_page(struct page *page, gfp_t gfp)
 {
 	dfprintk(PAGECACHE, "NFS: release_page(%p)\n", page);
 
+	if (gfp & __GFP_WAIT)
+		nfs_wb_page(page->mapping->host, page);
 	/* If PagePrivate() is set, then the page is not freeable */
 	if (PagePrivate(page))
 		return 0;
